src/tron/tronEngine.js

```diff
@@ -0,0 +1,548 @@
+      const networkFee = '0'
+      const nativeAmount = edgeSpendInfo.spendTargets[0].nativeAmount
+      const balanceTrx = this.walletLocalData.totalBalances[
+        this.currencyInfo.currencyCode
+      ]
+
+      if (bns.gt(nativeAmount, balanceTrx)) {
+        throw new InsufficientFundsError()
+      }
+
+      // Tron can only have one output
+      if (edgeSpendInfo.spendTargets.length !== 1) {
+        throw new Error('Error: only one output allowed')
+      }
+
+      const spendTarget = edgeSpendInfo.spendTargets[0]
+      const publicAddress = spendTarget.publicAddress
+      const data =
+        spendTarget.otherParams != null ? spendTarget.otherParams.data : void 0
+
+      let otherParams: Object = {}
+      // if TRON
+      if (currencyCode === PRIMARY_CURRENCY) {
+        // builds tx, but does NOT send it
+        const trxParams: TronTxOtherParams = await tronWeb.transactionBuilder.sendTrx(
+          publicAddress,
+          nativeAmount,
+          this.walletLocalData.publicKey
+        )
+        otherParams = trxParams
+        // if Token
+      } else {
+        let contractAddress = ''
+        if (data) {
+          contractAddress = publicAddress
+        } else {
+          const tokenInfo = this.getTokenInfo(currencyCode)
+          if (!tokenInfo || typeof tokenInfo.contractAddress !== 'string') {
+            throw new Error(
+              'Error: Token not supported or invalid contract address'
+            )
+          }
+
+          contractAddress = tokenInfo.contractAddress
+        }
+
+        // for Tokens:
+        // builds tx, but does NOT send it
+        const trxParams: TronTxOtherParams = await tronWeb.transactionBuilder.sendToken(
+          publicAddress,
+          nativeAmount,
+          contractAddress,
+          this.walletLocalData.publicKey
+        )
+        otherParams = trxParams
+      }
+      const edgeNativeAmount = bns.mul(nativeAmount, '-1')
+
+      // **********************************
+      // Create the unsigned EdgeTransaction
+
+      const edgeTransaction: EdgeTransaction = {
+        txid: '', // txid
+        date: 0, // date
+        currencyCode, // currencyCode
+        blockHeight: 0, // blockHeight
+        nativeAmount: edgeNativeAmount, // nativeAmount
+        networkFee, // networkFee, supposedly fixed
+        ourReceiveAddresses: [], // ourReceiveAddresses
+        signedTx: '', // signedTx
+        otherParams // otherParams
+      }
+      return edgeTransaction
+    } catch (e) {
+      this.log('makeSpend error: ', e)
+    }
+  }
+
+  async signTx (edgeTransaction: EdgeTransaction): Promise<EdgeTransaction> {
+    const privKey = this.walletInfo.keys.tronKey
+
+    const signedTx = await tronWeb.trx.sign(
+      edgeTransaction.otherParams,
+      privKey
+    )
+    // this.log(`SUCCESS TRX broadcastTx\n${JSON.stringify(signedTx, null, 2)}`)
+    edgeTransaction.signedTx = signedTx
```

Comment:
Please make sure that this is a string and not an object. The user interface will show this value if it exists and the user taps on the advanced details

Reply:
When changing from Object to String using JSON.stringify(signedTx, null,2), it fails to send a tx:
makeSpend error:  Invalid amount provided

I could successfully copy the 'signed raw hex' when clicking on 'view advanced transaction data' in the transaction details screen when it's an Object, if you meant that.

    edgeTransaction.signedTx = signedTx
VS
    edgeTransaction.signedTx = JSON.stringify(signedTx, null, 2)

Maybe I misunderstood something